arch/arm/include/kernel_arch_func.h

```diff
@@ -68,6 +68,16 @@ _arch_switch_to_main_thread(struct k_thread *main_thread,
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 
+#ifdef CONFIG_MPU_STACK_GUARD
```

Comment:
any reason this needs to be an asm, done here, vs moving it before the asm block?

Reply:
Doing it in asm lets me dance around a significant include file issue.

Reply:
Works for me